c7n/testing.py

```diff
@@ -95,6 +96,11 @@ def load_policy(
             config=config)
         # policy non schema validation is also lazy initialization
         [p.validate() for p in collection]
+        if not allow_deprecations:
+            for p in collection:
+                r = p.deprecation_report()
+                if r.has_deprecations:
+                    raise RuntimeError(f"policy {p.name} contains deprecated usage\n{r.format()}")
```

Comment:
I wasn't sure exactly if there was a better error to raise.

Reply:
PolicyValidationError is typical, see exceptions.py a dedicated exception class for this also sounds good.